render/render.js

```diff
@@ -101,13 +101,25 @@ module.exports = function($window) {
 		return element
 	}
 	function createComponent(parent, vnode, hooks, ns, nextSibling) {
-		vnode.state = Object.create(vnode.tag)
-		var view = vnode.tag.view
-		if (view.reentrantLock != null) return $emptyFragment
-		view.reentrantLock = true
-		initLifecycle(vnode.tag, vnode, hooks)
-		vnode.instance = Vnode.normalize(view.call(vnode.state, vnode))
-		view.reentrantLock = null
+		var sentinel
+		if (typeof vnode.tag === "function") {
+			vnode.state = null
+			sentinel = vnode.tag
+			if (sentinel.$$reentrantLock$$ != null) return $emptyFragment
+			sentinel.$$reentrantLock$$ = true
```

Comment:
More of a note for later, but @lhorie, is the re-entrant lock something we can pile into an array instead, to avoid mutating the component?

The reason I ask is because I'm not sure how deep the recursion tends to go from here. If the common case is just once or twice, the `indexOf` check would be just as cheap.